### PR TITLE
[WIP][🚨][RN] Updates RN version and changes WFU ListView to FlatList

### PR DESCRIPTION
**Pull request by sarahscott**
- [x] Updates to RN v0.43.0-rc.4
- [x] Updates our transformer to be compatible with the newer RN version
- [x] Converts `ListView` in Works For You container to `FlatList`
- [ ] Make sure it uses my branch of DefinitelyTyped with `FlatList` typings
- [ ] Fixes `InfiniteScrollArtworksGrid` bug in gene, and tests for artist and gene:

With the upgrade, the Gene container throws this error and doesn't load its artworks grid: 

>Warning: RelayContainer: component `InfiniteScrollArtworksGrid` was rendered with variables that differ from the variables used to fetch fragment `gene`. The fragment was fetched with variables `{"totalSize":10,"medium":"*","priceRange":"*-*","sort":"-partner_updated_at"}`, but rendered with variables `{"totalSize":10,"medium":"painting","priceRange":"50-10000","sort":"-partner_updated_at"}`. This can indicate one of two possibilities: 
> - The parent set the correct variables in the query - `InfiniteScrollArtworksGrid.getFragment('gene', {...})` - but did not pass the same variables when rendering the component. Be sure to tell the component what variables to use by passing them as props: `<InfiniteScrollArtworksGrid ... totalSize={...} medium={...} priceRange={...} sort={...} />`.
> - You are intentionally passing fake data to this component, in which case ignore this warning.

The Artist container does load its infinite scroll grid and doesn't throw any errors or warnings, but its tests aren't loading. This shouldn't be merged until we fix that up.

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>Please add a description to your PR.</td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>Please assign someone to merge this PR, and optionally include people who should review.</td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: package.json [hunk 1]
```diff
@@ -69,8 +69,8 @@
   ],
   "dependencies": {
     "lodash": "3.10.1",
-    "react": "latest",
-    "react-native": "https://github.com/alloy/react-native.git#v0.42.0-with-customisable-source-exts",
+    "react": "16.0.0-alpha.4",
+    "react-native": "file:../react-native",
```
Comment (sarahscott) [inline package.json]: I can change this to use the correct fork once I update the fork to v0.43.0-rc.4.

File: Example/Podfile [hunk 2]
```diff
@@ -12,7 +12,7 @@ target 'Emission' do
   pod 'Emission', :path => '../'
 
   # As this runs dev, we need the developer web socket
-  pod 'React', :path => react_path, :subspecs => %w(RCTWebSocket)
+  pod 'React', :path => react_path, :subspecs => %w(RCTWebSocket DevSupport)
```
Comment (sarahscott) [inline Example/Podfile]: See: https://github.com/facebook/react-native/issues/13127

File: src/lib/containers/works_for_you.tsx [hunk 3]
```diff
@@ -89,15 +81,18 @@ export class WorksForYou extends React.Component<Props, State> {
...
-    return(
-      <ListView dataSource={this.state.dataSource}
-                renderRow={(notification) => <Notification notification={notification}/>}
-                renderSeparator={(sectionID, rowID) =>
-                  <View key={`${sectionID}-${rowID}`} style={styles.separator} /> as React.ReactElement<{}>
-                }
-                style={{marginTop: this.state.topMargin}}
-      />)
+    const notifications = this.props.me.notifications_connection.edges.map((edge, index) => {
+      return { key: "notification-" + index, node: edge.node }
```
Comment (sarahscott) [inline src/lib/containers/works_for_you.tsx]: the `key` can be the `__id` once metaphysics is deployed with that change.

**Comment by sarahscott** [comment]
Going to close this now; I want to give `FlatList` a shot after upgrading our RN version.

**Pull request closed without merging**